registry/handlers/httperror.go

```diff
@@ -0,0 +1,34 @@
+package handlers
+
+import (
+	"github.com/docker/distribution/registry/api/v2"
+	"net/http"
+)
+
+// HTTPError is a wrapper for v2.Errors that adds a Status int to hold
+// an HTTP Status code which will be used to set the status code on
+// the response.
+type HTTPError struct {
+	v2.Errors
```

Comment:
Because you wanted to remove v2.Errors from Context and it has to go somewhere as the collecting of multiple errors into v2.Errors is in use already.